docs/DafnyRef/UserGuide.md

```diff
@@ -51,129 +96,315 @@ verification failure in a file that is being checked.
-default, but can be skipped by using the `-noVerify` or `--no-verify` option
-- Compiling the dafny program to a target language. This step is performed by default if the verification is
-successful but can be skipped or always executed by using variations of the `-compile` option or different commands.
-- Whether the source code of the compiled target is written out is controlled by `-spillTargetCode` or the top-level command that is being executed.
-- The particular target language used is chosen by `-compileTarget` (old-style) or `--target` (new-style).
+The command executes the `dafny resolve` phase (accepting its options) and has the following additional options:
+
+- `--report-file:<report-file>` --- spcifies the path where the audit
+   report file will be stored. Without this option, the report
+    will be issued as standard warnings, written to standard-out.
+- `--report-format:<format>` --- specifies the file format to use for
+   the audit report. Supported options include: 
+   - 'txt, 'text': plain text in the format of warnings
+   - 'html': standalone HTML ('html')
+   - 'md', 'markdown', 'md-table', 'markdown-table': a Markdown table
+   - 'md-ietf', 'markdown-ietf': an IETF-language document in Markdown format
+   - The default is to infer the format from the filename extension
+- `--compare-report` --- compare the report that would have
+   been generated with the existing file given by --report-file, and fail if
+   they differ.
+
+The command emits exit codes of TODO.
+
+#### 25.5.1.9. `dafny test` {#sec-dafny-test}
+ 
+This _experimental_ command (verifies and compiles the program and) runs every method in the program that is annotated with the `{:test}` attribute.
+Verification can be disabled using the `--no-verify` option. `dafny test` also accepts all other options of the `dafny build` command. 
+In particular, it accepts the `--target` option that specifies the programming language used in the build and execution phases.
+
+<!-- TODO: Or is it like dafny run? -->
+
+There are currently no other options specific to the `dafny test` command.
+
+The order in which the tests are run is not specified.
+
+For example, this code (as the file `t.dfy`)
+```
+method {:test} m() {
+  mm();
+  print "Hi!\n";
+}
+
+method mm() {
+  print "mm\n";
+}
+
+module M {
+  method {:test} q() {
+    print 42, "\n";
+  }
+}
+
+class A {
+  static method {:test} t() { print "T\n"; }
+}
+```
+and this command-line
+```bash
+dafny test --no-verify t.dfy
+```
+produce this output text:
+```text
+M.q: 42
+PASSED
+A.t: T
+PASSED
+m: mm
+Hi!
+PASSED
+```
+
+#### 25.5.1.10. `dafny generate-tests` {#sec-dafny-generate-tests}
+
+This _experimental_ command (verifies the program and) then generates unit test code (as Dafny source code) that provides
+complete coverage of the method.
+
+Such methods must be static and have no input parameters.
+
+_This command is under development and not yet functional._
```

Comment:
Under development is already indicated by this command being experimental. What do you mean by not yet functional? It can be used.

Reply:
I see. I think if it doesn't work at all the command shouldn't be in the list. Thanks for following up!